### PR TITLE
add a health endpoint

### PR DESCRIPTION
**Pull request by andibraeu**
I use openhab cloud in a kubernetes environment. When the mongodb connection gets lost, it doesn't recover alone. 

Kubernetes doesn't see that the state isn't healthy by just probing the home page. So I created a health endpoint that returns HTTP status 200 in case every is ok, else it returns HTTP status 500.

### CODE REVIEW

**Comment by andibraeu** [comment]
@digitaldan can you please have a look on that?

**Comment by digitaldan** [comment]
Hi, LGTM, i'm working in the code this week, so will give this a quick test today or tomorrow. Thanks!

**Comment by digitaldan** [comment]
HI one comment, can you add a config option to enable/disable this?  I don't want to have this exposed on our public service, and would prefer not to try and block this specific route in our load balancer (its already complicated).  Thanks !

**Comment by andibraeu** [comment]
Sure, I added a config option. Usually things like that I would only expose to another endpoint, that is not public.

**Comment by andibraeu** [comment]
@digitaldan can you please take a look again?

File: routes/health.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        } else {
+            return res.status(500).json({
+                status: "Not OK",
+                mongoose: mongoose_state,
+                errors: errors
+            });
+        }
+
+        function collectErrors() {
+            var errors = [];
```
Comment (digitaldan) [inline routes/health.js:30]: Curious why this returns an array, it looks like to me this can ever have one value?   Why not return the error itself ?

**Comment by digitaldan** [comment]
Hi, sorry i missed your changes in my sea of github notifications.  Just left a few questions.

File: routes/health.js (hunk 1 above)
Comment (andibraeu) [reply routes/health.js:30]: At the moment the health endpoint just checks for mongodb connection errors. I thought of adding more checks later, for example for redis or some of the integration like ifttt.

File: deployment/docker-compose/config.json.template [hunk 2]
```diff
@@ -6,7 +6,10 @@
       "logger" : {
         "type": "console"
       },
-      "subDomainCookies": false
+      "subDomainCookies": false,
+      "healthEndpoint" : {
```
[the remark this answers is not in the document]
Comment (andibraeu) [reply deployment/docker-compose/config.json.template:10]: yes, for example we could add another option to configure not to return error details, but just the health status.
Or configure, as I said in the other answer to check for other modules/dependencies.

File: routes/health.js (hunk 1 above)
Comment (digitaldan) [reply routes/health.js:30]: Got it, redis sounds like a good candidate.

**Comment by digitaldan** [comment]
LGTM, thanks.

**Pull request merged**